src/BccCode.Linq/ApiClient/ResultList.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System.Collections;
+using System.ComponentModel.DataAnnotations;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+
+namespace BccCode.ApiClient;
+
+[DataContract(Name = "ResultList")]
+public abstract class ResultList
+{
+    protected ResultList(Metadata? meta = null)
+    {
+    }
+
+    /// <summary>
+    /// Gets or Sets Meta
+    /// </summary>
+    [DataMember(Name = "meta", EmitDefaultValue = false)]
+    public Metadata? Meta { get; set; }
+
+    [JsonIgnore]
+    public IEnumerable<object>? Data { get; set; }
+
+    public abstract void AddData(IEnumerable? data);
+
+    [Obsolete($"Use property {nameof(Data)} instead")]
```

Comment:
Little unsure why this is obsolete. The idea is that IEnumerable<T> is on the derived class, not on the base class. The base class type is used for retreving the data in scenarios where the type of the collection is unknown (e.g. when merging results from multiple requests in a generic handler).

Reply:
In general, I'd say the ResultList should be designed unmutable (read-only). This would give us the possibility in the future to use better optimized types like `ImmutableDictionary<,>`. See https://stackoverflow.com/questions/16612936/immutable-dictionary-vs-dictionary-vs-c5 as well.

I do not see the reason people should change the `ResultList`. We only need that internally in the client SDK and there I think we could use reflection if we really need that.

Reply:
Will all merging of results from multiple paged requests happen within the BccCode.Linq library? I'm not really in favour of using reflection unless there's a need for it. Instead we could rather have an additional ImmutableResultList and create that from a result list if necessary. I.e. list.ToImmutable()

Reply:
Immutable is now implemented in the style like C# does it:
* all immutables are moved into an extra namespace: `BccCode.ApiClient.Immutable`
* there exist an extension method `((IResultList<T>)resultList).ToImmutableResutList` and `((IMetadata)metadata).ToImmutableMetadata()`

The immutable code probably could be optimized but I think the interface how to use it is now clear so I will leave it for now like this.